GVFS/GVFS.Service/GVFSMountProcess.Windows.cs

```diff
@@ -6,65 +6,57 @@
 
 namespace GVFS.Service
 {
-    public class GVFSMountProcess : IDisposable
+    public class GVFSMountProcess : IRepoMountProcess
     {
-        private const string ParamPrefix = "--";
-
-        private readonly ITracer tracer;
-
-        public GVFSMountProcess(ITracer tracer, int sessionId)
-        {
-            this.tracer = tracer;
-            this.CurrentUser = new CurrentUser(this.tracer, sessionId);
-        }
-
-        public CurrentUser CurrentUser { get; private set; }
-
-        public bool Mount(string repoRoot)
+        public bool MountRepository(string repoRoot, int sessionId, ITracer tracer)
         {
-            if (!ProjFSFilter.IsServiceRunning(this.tracer))
+            if (!ProjFSFilter.IsServiceRunning(tracer))
             {
                 string error;
-                if (!EnableAndAttachProjFSHandler.TryEnablePrjFlt(this.tracer, out error))
+                if (!EnableAndAttachProjFSHandler.TryEnablePrjFlt(tracer, out error))
                 {
-                    this.tracer.RelatedError($"{nameof(this.Mount)}: Could not enable PrjFlt: {error}");
+                    tracer.RelatedError($"{nameof(this.MountRepository)}: Could not enable PrjFlt: {error}");
+                    return false;
                 }
             }
 
-            if (!this.CallGVFSMount(repoRoot))
+            using (CurrentUser currentUser = new CurrentUser(tracer, sessionId))
             {
-                this.tracer.RelatedError($"{nameof(this.Mount)}: Unable to start the GVFS.exe process.");
-                return false;
-            }
+                if (!this.CallGVFSMount(repoRoot, currentUser))
+                {
+                    tracer.RelatedError($"{nameof(this.MountRepository)}: Unable to start the GVFS.exe process.");
+                    return false;
+                }
 
-            string errorMessage;
-            if (!GVFSEnlistment.WaitUntilMounted(repoRoot, false, out errorMessage))
-            {
-                this.tracer.RelatedError(errorMessage);
-                return false;
+                string errorMessage;
+                if (!this.WaitUntilMounted(repoRoot, false, out errorMessage))
+                {
+                    tracer.RelatedError(errorMessage);
+                    currentUser.Dispose();

```

Comment:
Isn't this `Dispose()` implicitly covered by the `using` statement?

Reply:
Yes. Removed it.